metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorDatabaseService.java

```diff
@@ -122,7 +128,11 @@ public boolean exists(@Nonnull final ConnectorContext requestContext, @Nonnull f
         try {
             List<QualifiedName> qualifiedNames = Lists.newArrayList();
             for (String databaseName : metacatHiveClient.getAllDatabases()) {
-                qualifiedNames.add(QualifiedName.ofDatabase(name.getCatalogName(), databaseName));
+                final QualifiedName qualifiedName = QualifiedName.ofDatabase(name.getCatalogName(), databaseName);
+                if (prefix != null && !qualifiedName.toString().startsWith(prefix.toString())) {
```

Comment:
This is somewhat innefficiant cause you're basically making a string comparison on a longer string `catalogName/databasePrefix` vs just comparing the prefix against the database name. I'm not sure how we want do do it but rather than doing this comparison over and over I would just once outside the loop compare catalogs and make sure they match and then only compare the actual database names in here.